apps/system/js/utility_tray.js

```diff
@@ -437,49 +438,39 @@ window.UtilityTray = {
     if (Service.query('locked')) {
       this.hide(true);
     } else {
-      var timeDelta = timestamp - this.lastMoveTime;
-      var significant = (Math.abs(this.lastDelta) > (this.screenHeight / 5));
-      var shouldOpen = significant ? !this.shown : this.shown;
-
-      if (significant && timeDelta <= this.MAX_SWIPE_AGE &&
-          (this.lastMove < 0) === this.shown) {
-        var velocity = (timeDelta / 1000) / Math.abs(this.lastMove);
-        this.animationTime =
-          Math.min(this.DEFAULT_ANIMATION_TIME,
-            Math.max(this.MINIMUM_ANIMATION_TIME,
-              velocity * (this.shown ?
-                          this.screenHeight + this.lastDelta :
-                          this.screenHeight - this.lastDelta)));
-      }
 
-      shouldOpen ? this.show() : this.hide();
-    }
+      var endY = touch.pageY;
+      var down = endY > this.startY;
 
-    /*
-     * Trigger search from the left half of the screen if we're LTR
-     * And trigger from the right half if we're RTL.
-     */
-    var corner;
-    if (document.documentElement.dir  == 'rtl') {
-      corner = touch && (touch.target === this.topPanel) &&
-                 (touch.pageX > (window.innerWidth / 2));
-    } else {
-      corner = touch && (touch.target === this.topPanel) &&
-                 (touch.pageX < (window.innerWidth / 2));
-    }
-    if (this.isTap && corner) {
-      if (this.showing) {
-        this.hide();
-      }
-      var app = Service.query('getTopMostWindow');
-      var combinedView = app.appChrome && app.appChrome.useCombinedChrome();
-      var isTransitioning = app.isTransitioning();
+      // We don't do any animation when the tray
+      // is down and the user swipes down.
+      var ignore = this.shown && down;
 
-      if (!isTransitioning && combinedView && !app.appChrome.isMaximized()) {
-        app.appChrome.titleClicked();
+      var timeDelta = timestamp - this.lastMoveTime;
+      var significant = (Math.abs(this.lastDelta) > (this.screenHeight / 5));
+      var shouldShow = significant ? !this.shown : this.shown;
+      var action = shouldShow ? 'show' : 'hide';
+
+      this.transitioning = false;
+
+      if (!ignore) {
+        if (action && timeDelta <= this.MAX_SWIPE_AGE &&
+            (this.lastMove < 0) === this.shown) {
+          var velocity = (timeDelta / 1000) / Math.abs(this.lastMove);
+          this.animationTime =
+            Math.min(this.DEFAULT_ANIMATION_TIME,
+              Math.max(this.MINIMUM_ANIMATION_TIME,
+                velocity * (this.shown ?
+                            this.screenHeight + this.lastDelta :
+                            this.screenHeight - this.lastDelta)));
```

Comment:
I think I got STRs:
* launch contact
* tap "+"
* focus a field, wait for the keyboard
* tap "Search the web", wait for the rocketbar overlay
* close the rocketbar

-> unable to show the utility tray and
```
W/GeckoConsole( 1710): [JavaScript Warning: "Error in parsing value for 'transition-duration'.  Declaration dropped." {file: "app://system.gaiamobile.org/index.html" line: 0 column: 0 source: "-2.766089108910874ms"}]
```
in the console